File: packages/gbnf-experiment/data/20260909T005215Z_9a3e20fe/ported_implementation/src/utils/errors/buildErrorPosition.ts
export const MAXIMUM_NUMBER_OF_ERROR_LINES_TO_SHOW = 3;

export const buildErrorPosition = (src: string, pos: number): string[] => {
  if (src === '') {
    return ['No input provided'];
  }
  const lines = src.split('\n');

  let lineIdx = 0;
  let cursor = pos;
  while (lines[lineIdx] && cursor > lines[lineIdx].length - 1 && cursor < src.length) {
    cursor -= lines[lineIdx].length;
    lineIdx += 1;
  }

  const linesToShow = lines.slice(
    Math.max(0, lineIdx - (MAXIMUM_NUMBER_OF_ERROR_LINES_TO_SHOW - 1)),
    lineIdx + 1,
  );

  return [...linesToShow, `${' '.repeat(Math.max(0, cursor))}^`];
};

File: proposed-projects/round-trip-experiment/reverse/20260909T005215Z_9a3e20fe/20260910T045026Z_2a40c4a6/ported_implementation/gbnf/utils/errors/build_error_position.py
MAXIMUM_NUMBER_OF_ERROR_LINES_TO_SHOW = 3


def build_error_position(src: str, pos: int) -> list[str]:
    if src == "":
        return ["No input provided"]
    lines = src.split("\n")

    line_idx = 0
    cursor = pos
    while (
        line_idx < len(lines)
        and lines[line_idx]
        and cursor > len(lines[line_idx]) - 1
        and cursor < len(src)
    ):
        cursor -= len(lines[line_idx])
        line_idx += 1

    lines_to_show = lines[
        max(0, line_idx - (MAXIMUM_NUMBER_OF_ERROR_LINES_TO_SHOW - 1)) : line_idx + 1
    ]

    return [*lines_to_show, f"{' ' * max(0, cursor)}^"]
